### PR TITLE
[Feature/extensions] Renaming tests to be consistent with main branch, changing latches, and removing NamedWriteableRegistry from extensions

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Renaming tests to be consistent with main branch

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ryanbogan** [comment]
The test that is failing is also broken on main - I will rebase once the fix is pushed there.

**Comment by dreamer-89** [comment]
Last few gradle check failures happening due to version bump after 2.4.1 release, https://github.com/opensearch-project/OpenSearch/pull/5560. Fixed is merged into main branch. @ryanbogan : Can you please rebase your branch against latest `main` branch ?
```
* What went wrong:
Execution failed for task ':distribution:bwc:bugfix:buildBwcLinuxTar'.
> Building 2.4.1 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/bugfix/build/bwc/checkout-2.4/distribution/archives/linux-tar/build/distributions/opensearch-min-2.4.1-SNAPSHOT-linux-x64.tar.gz
```

**Review by dbwiddis — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java [hunk 1]
```diff
@@ -175,7 +175,7 @@ public String executor() {
                 restExecuteOnExtensionResponseHandler
             );
             try {
-                inProgressLatch.await(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+                inProgressFuture.get(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
```
[the remark this answers is not in the document]
Comment (ryanbogan) [reply server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java:178]: @dbwiddis I wanted this to still be synchronous for now.  I think we should either raise a PR on feature/extensions that changes all the latches and `.get()` calls or we should wait and do a similar PR once everything is merged to main.  Either way, I think this approach is better than changing them one by one as we merge everything to main.  Thoughts?

[the remark this answers is not in the document]
Comment (dbwiddis) [reply server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java:178]: Fair, raise an issue, add a TODO comment and we can merge this.

[the remark this answers is not in the document]
Comment (owaiskazi19) [reply server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java:179]: @ryanbogan can you link the issue here?

[the remark this answers is not in the document]
Comment (ryanbogan) [reply server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java:179]: https://github.com/opensearch-project/OpenSearch/issues/5595

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**